eng/build.sh

```diff
@@ -246,7 +247,7 @@ function BuildSolution {
   # NuGet often exceeds the limit of open files on Mac and Linux
   # https://github.com/NuGet/Home/issues/2163
   if [[ "$UNAME" == "Darwin" || "$UNAME" == "Linux" ]]; then
-    ulimit -n 6500
+    ulimit -n 6500 || echo "Cannot change ulimit"
```

Comment:
Unsure why but `ulimit` was failing for me in my local development. I'm using WSL2 and couldn't find any specific docs on why this was broken but did find a number of people reporting it. It didn't feel like a truly critical failure though so I added this condition. 

Happy to do something better here if such a item exists.